pkg/logging/fields.go

```diff
@@ -32,90 +39,123 @@ var resourceTypeFields = map[string]string{
 	notifierField:     administrationResources.Notifier,
 }
 
+type ContextField struct {
+	fields []zap.Field
+}
+
+func (c *ContextField) add(field zap.Field) {
+	c.fields = append(c.fields, field)
+}
+
+func (c *ContextField) MarshalLogObject(enc zapcore.ObjectEncoder) error {
+	for _, field := range c.fields {
+		field.AddTo(enc)
+	}
+	return nil
+}
+
+// Context provides selected context values as structured log fields.
+func Context(ctx context.Context) zap.Field {
```

Comment:
I think this name is missleading. Compare with `trace.Context` that creates a context here we get field from context so it should be `FieldFromContext`
nit: Should we return a slice of fields just in case we will add something in a future?
ref: 
- https://github.com/uber-go/zap/issues/654

Reply:
How about `FromContext`?

I didn't return a slice on purpose and instead introduced the `ContextField` type as a workaround. The slice approach has the problem that it messes up the type inference for `Errorw(msg string, keysAndValues ...interface{})` because of the `interface{}`.

```
log.Errorw("Failed to acquire scan semaphore",
  logging.FromContext(ctx), // This is type []zap.Field.
  logging.ImageName(request.GetImage().GetName().GetFullName()), // This is type zap.Field.
  logging.ImageID(request.GetImage().GetId()),
  logging.Err(err),
)
```

I solved this issue by having `ContextField` hold the field slice internally and bundle it up via `zap.Inline`.